### PR TITLE
add some logging for tensorify

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143391

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by ezyang** [comment]
deferring to @jamesjwu

File: torch/_dynamo/metrics_context.py [hunk 1]
```diff
@@ -67,14 +67,14 @@ def increment(self, metric: str, value: int) -> None:
             self._metrics[metric] = 0
         self._metrics[metric] += value
 
-    def set(self, metric: str, value: Any) -> None:
+    def set(self, metric: str, value: Any, overwrite: bool = False) -> None:
```
Comment (jamesjwu) [inline torch/_dynamo/metrics_context.py:70]: @ezyang you didn't like this when I added it before, did you? Unlike the remote cache case though I don't really know if there's an alternative here.

**Review by jamesjwu — APPROVED** [review]
LGTM otherwise, @masnesral, mind giving your opinion on the `overwrite` change rq too

File: torch/_dynamo/metrics_context.py (hunk 1 above)
Comment (masnesral) [reply torch/_dynamo/metrics_context.py:70]: I actually don't remember the argument against `overwrite`. I guess in general we wouldn't expect individual values in the metrics context to change within the same context. If it's valid for them to change, then maybe we'd be losing some detail by overriding with the last value. I don't have the context for these new fields. Why is it expected they might change and we'd only want to log the last-seen value?

Comment (ezyang) [reply torch/_dynamo/metrics_context.py:70]: Yeah, I didn't like it. Is there an aggregate we can compute instead? Even a list append?

**Pull request closed without merging**